utilities/rpp-unittests/SOW3_OCL/BatchPD_ocl_pln3.cpp

```diff
@@ -0,0 +1,880 @@
+        inputf32_secondTemp = inputf32_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputf32Temp = ((Rpp32f)*inputTemp) / 255.0;
+            *inputf32_secondTemp = ((Rpp32f)*input_secondTemp) / 255.0;
+            inputTemp++;
+            inputf32Temp++;
+            input_secondTemp++;
+            inputf32_secondTemp++;
+        }
+    }
+    else if (ip_bitDepth == 5)
+    {
+        Rpp8u *inputTemp, *input_secondTemp;
+        Rpp8s *inputi8Temp, *inputi8_secondTemp;
+
+        inputTemp = input;
+        input_secondTemp = input_second;
+
+        inputi8Temp = inputi8;
+        inputi8_secondTemp = inputi8_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputi8Temp = (Rpp8s)(((Rpp32s)*inputTemp) - 128);
+            *inputi8_secondTemp = (Rpp8s)(((Rpp32s)*input_secondTemp) - 128);
+            inputTemp++;
+            inputi8Temp++;
+            input_secondTemp++;
+            inputi8_secondTemp++;
+        }
+    }
+
+    cl_mem d_input, d_inputf16, d_inputf32, d_inputi8, d_input_second, d_output, d_outputf16, d_outputf32, d_outputi8;
+    cl_platform_id platform_id;
+    cl_device_id device_id;
+    cl_context theContext;
+    cl_command_queue theQueue;
+    cl_int err;
+    err = clGetPlatformIDs(1, &platform_id, NULL);
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_GPU, 1, &device_id, NULL);
+    theContext = clCreateContext(0, 1, &device_id, NULL, NULL, &err);
+    theQueue = clCreateCommandQueue(theContext, device_id, 0, &err);
+    d_input = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+    d_inputf16 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp16f), NULL, NULL);
+    d_inputf32 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp32f), NULL, NULL);
+    d_inputi8 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8s), NULL, NULL);
+    d_input_second = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+    d_output = clCreateBuffer(theContext, CL_MEM_READ_ONLY, oBufferSize * sizeof(Rpp8u), NULL, NULL);
+    d_outputf16 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, oBufferSize * sizeof(Rpp16f), NULL, NULL);
+    d_outputf32 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, oBufferSize * sizeof(Rpp32f), NULL, NULL);
+    d_outputi8 = clCreateBuffer(theContext, CL_MEM_READ_ONLY, oBufferSize * sizeof(Rpp8s), NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_input, CL_TRUE, 0, ioBufferSize * sizeof(Rpp8u), input, 0, NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_inputf16, CL_TRUE, 0, ioBufferSize * sizeof(Rpp16f), inputf16, 0, NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_inputf32, CL_TRUE, 0, ioBufferSize * sizeof(Rpp32f), inputf32, 0, NULL, NULL);
+    err = clEnqueueWriteBuffer(theQueue, d_inputi8, CL_TRUE, 0, ioBufferSize * sizeof(Rpp8s), inputi8, 0, NULL, NULL);
```

Comment:
Codacy found an issue: [Variable 'err' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)